core/modules/widgets/button.js

```diff
@@ -203,35 +203,72 @@ ButtonWidget.prototype.setTiddler = function() {
 	}
 };
 
+/*
+Collect the attributes we need, in the process determining whether we're being used in legacy mode
+*/
+ButtonWidget.prototype.collectAttributes = function() {
+	// Detect legacy mode: true if no attributes start with $
+	this.legacyMode = this.isLegacyMode();
+	// Get attributes for the appropriate mode
+	if(this.legacyMode) {
```

Comment:
What I do not like is, that we already know, if legacy mode is active or not. So the lines of the if branch and the else branch are exactly the same code. Only the $ is different. 

`this.legacyMode` is already defined. So we may use that info in `this.getAttribte()` so we can avoid code duplication. 

We only need to make sure, that `this.legacyModo` is valid 

Just some thoughts.